### PR TITLE
Add button for zwave_js options flow

### PR DESCRIPTION
**Pull request by MartinHjelmare**
## Proposed change

- Add a button "Re-configure Server" to start the zwave_js options flow.
- This requires a PR in the Core repo: https://github.com/home-assistant/core/pull/51840

zwave_js_options_flow_button_desktop_ok

zwave_js_options_flow_button_mobile_ok

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-config-dashboard.ts [hunk 1]
```diff
@@ -269,6 +276,18 @@ class ZWaveJSConfigDashboard extends LitElement {
     );
   }
 
+  private async _openOptionFlow() {
+    const searchParams = new URLSearchParams(window.location.search);
+    if (!searchParams.has("config_entry")) {
```
Comment (cgarwood) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-config-dashboard.ts:281]: `this.configEntryId` should be passed in from the router so we shouldn't need to pull it from the URL search params here

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-config-dashboard.ts [hunk 2]
```diff
@@ -269,6 +276,18 @@ class ZWaveJSConfigDashboard extends LitElement {
...
+      return;
```
Comment (cgarwood) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-config-dashboard.ts:282]: Can guard instead with `if (!this.configEntryId) { return; }`

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-config-dashboard.ts [hunk 3]
```diff
@@ -262,6 +269,17 @@ class ZWaveJSConfigDashboard extends LitElement {
     );
   }
 
+  private async _openOptionFlow() {
+    if (!this.configEntryId) {
+      return;
+    }
+    const configEntries = await getConfigEntries(this.hass);
+    const configEntry = configEntries.find(
+      (entry) => entry.entry_id === this.configEntryId!
```
Comment (ludeeus) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-config-dashboard.ts:278]:
```suggestion
      (entry) => entry.entry_id === this.configEntryId
```

You already check if this exists so `!`  it's not needed here, right?

File: src/translations/en.json [hunk 4]
```diff
@@ -2586,7 +2586,8 @@
             "home_id": "Home ID",
             "close": "Close",
             "add_node": "Add Node",
-            "remove_node": "Remove Node"
+            "remove_node": "Remove Node",
+            "reconfigure_server": "Re-configure Server"
```
Comment (ludeeus) [inline src/translations/en.json:2590]: It is the integration (config entry) you re-configure and not the server?
Is this naming on purpose?

Comment (MartinHjelmare) [reply src/translations/en.json:2590]: It'll change the server options both in the config entry and, if used, in the add-on.

**Comment by MartinHjelmare** [comment]
Backend PR is merged.

**Review by ludeeus — APPROVED** [review]

**Pull request merged**